papas/reconstruction/src/PFBlock.cpp

```diff
@@ -27,17 +27,17 @@ bool blockIdComparer(Identifier id1, Identifier id2) {
 PFBlock::PFBlock(const Ids& element_ids, Edges& edges, unsigned int index, char subtype)
     : m_id(IdCoder::makeId(index, IdCoder::kBlock, subtype, element_ids.size())), m_elementIds(element_ids) {
   PFBlock::tempBlockCount += 1;
-  m_elementIds.sort(blockIdComparer);
   // extract the relevant parts of the complete set of edges and store this within the block
   // note the edges will be removed from the edges unordered_map
   for (auto id1 : m_elementIds) {
     for (auto id2 : m_elementIds) {
       if (id1 >= id2) continue;
       // move the edge from one unordered map to the other
-      auto e = edges.find(Edge::makeKey(id1, id2));
-      if (e == edges.end()) throw std::range_error("Required Edge is missing from edges collection");
-      m_edges.emplace(e->second.key(), std::move(e->second));
-      edges.erase(e);
+      const auto& e = edges.find(Edge::makeKey(id1, id2));
+      if (e != edges.end()) {
```

Comment:
Hi Colin 
This is the unordered_map find.
I check in the following line to see if it was found ( it equals edges.end() if not found).
If the edge is not found its not added to the PFBlock edges collection. 
Papas should work fine provided we have an edge for everything that is linked. In fact, as we only ever remove links we could probably simplify this even further, and only have an edge if there is a link, in which case an edge need only contain the distance... A little more thought needed though.